### ISSUE TITLE
Allow @ConfigurationProperty to apply to class members when using @ConfigProperties

### ISSUE THREAD

**Issue by GregJohnStewart** [quarkusio/quarkus#7220]
(version `1.2.0.Final`, using Gradle)

**Description**
As it stands, I can't seem to properly apply `@ConfigurationProperty` annotations to members of a class that has the `@ConfigurationProperties` on it. Current behavior it seems is that the members' annotations are ignored. I noticed that the annotation works for interfaces, but I would like it to work for classes as well.

Example:

```
@Data
@AllArgsConstructor
@NoArgsConstructor
@ConfigProperties(prefix = "runningInfo")
public class ServerInfoBean {
    @ConfigProperty(name = "organization")
    String organization;
    @ConfigProperty(name = "serverName")
    private String serverName;
    @ConfigProperty(name = "url")
    private String url;
}
```

Error I get indicating annotations are ignored:

```
DeploymentException: No config value of type [java.lang.String] exists for: runningInfo.server-name
```

**Comment by geoand** [comment]
I think this is a very reasonable request. I would definitely be willing to help anyone interested in implementing it :)

**Comment by GregJohnStewart** [comment]
I'd love to try my hand at implementing, or at least help :) I would just ask for pointers on where to look and start, potentially haha

**Comment by geoand** [comment]
Great! I'll provide some pointers tomorrow morning

**Comment by geoand** [comment]
The basic place that needs to be updated is: https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java#L202

Currently for each field we just use its name to determine the property that will be used to back it. However that needs to change so that for each field we first check if the field has a `@ConfigProperty` annotation (although we will also have to decide how to handle the `defaultValue` field of the annotation).
For reference, here is how `@ConfigProperty` is handled for interfaces: https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/InterfaceConfigPropertiesUtil.java#L114

There is also another thing that will need to be taken care of - the making Quarkus not fail with the `DeploymentException: No config value of type [java.lang.String] exists for:` error. But that can be left as a second task - and I might just end up pushing a commit to the PR :)

**Comment by GregJohnStewart** [comment]
Looking at the code, it's more than I expected haha.

As a comment on that last bit, i would argue that maybe it should fail, at least when not using `Optional`, since its supported. Maybe a field in the `@ConfigProperty` that marks it as required/ not?

**Comment by geoand** [comment]
There is a lot going on indeed, but we need to generate a class based on a lot of metadata.

It indeed should fail in the same way that it would fail if used on the interface.

**Comment by geoand** [comment]
This was fixed in Quarkus 1.8
